### PR TITLE
Update master branch to v2.0.0

### PR DESCRIPTION
**Pull request by danforbes**
Relates to #79 and #80. I am also working on updating the `workshop` branch to v2.0.0

CC: @JoshOrndorff

### CODE REVIEW

**Comment by danforbes** [comment]
Marking the 2.0 versions as drafts because it seems that when they are run in development mode they do not produce blocks 🤔

File: README.md [hunk 1]
```diff
@@ -48,7 +50,7 @@ The following example takes you through a scenario where:
 
 1. Compile and build a release node
 ```bash
-cargo build --release
+cargo +nightly-2020-10-05 build --release
```
Comment (JoshOrndorff) [inline README.md:53]: Consider using a `rust-toolchain` file.

docs: https://rust-lang.github.io/rustup/overrides.html?highlight=rust-toolchain#the-toolchain-file
example: https://github.com/PureStake/moonbeam/blob/master/rust-toolchain

File: node/src/service.rs [hunk 2]
```diff
@@ -197,7 +195,7 @@ pub fn new_light(config: Configuration, sr25519_public_key: sr25519::Public) ->
 		client.clone(),
 		Sha3Algorithm::new(client.clone()),
 		0, // check inherents starting at block 0
-		Some(select_chain),
+		select_chain,
 		inherent_data_providers.clone(),
 		// TODO what should I Really use here? This compiles, and the `can_author_with` from line 196 doesn't??
```
Comment (JoshOrndorff) [inline node/src/service.rs:200]: Is "line 196" still a valid reference?

**Review by JoshOrndorff — COMMENTED** [review]
Looks good after just looking it over. The removal of the `mine` function was a surprise, but I think I remember that changing in the recipes also right?

File: node/src/service.rs (hunk 2 above)
Comment (apopiak) [inline node/src/service.rs:200]:
```suggestion
		// TODO what should I Really use here? This compiles, and the `can_author_with` from above doesn't??
```

**Comment by apopiak** [comment]
> The removal of the `mine` function was a surprise, but I think I remember that changing in the recipes also right?

I have this question, too, LGTM otherwise

**Comment by danforbes** [comment]
@JoshOrndorff & @apopiak - `mine` is not in the trait in v2.0.0 🤷🏻 https://substrate.dev/rustdocs/v2.0.0/sc_consensus_pow/trait.PowAlgorithm.html

Regarding the fact that blocks are not being authored in `--dev` mode, Joshy suggested the following:

> What if you add `--validator`? If that solves it then there is some easy code we could add to service.rs to fix `--dev` mode.

**Comment by apopiak** [comment]
When I add `--validator` or `--alice`  it still does not seem to produce blocks.

**Comment by apopiak** [comment]
The miner is started but does not seem to do anything useful?

File: node/src/service.rs [hunk 3]
```diff
@@ -153,19 +153,17 @@ pub fn new_full(config: Configuration, sr25519_public_key: sr25519::Public) -> R
 		let can_author_with =
 			sp_consensus::CanAuthorWithNativeVersion::new(client.executor().clone());
 
-		sc_consensus_pow::start_mine(
+		sc_consensus_pow::start_mining_worker(
```
Comment (apopiak) [inline node/src/service.rs:156]: This is started, but does not seem to actually mine anything.

**Comment by jimmychu0807** [comment]
Closing it as it has been a while.

**Pull request closed without merging**